shared/chat/conversation/messages/system/index.js

```diff
@@ -261,6 +261,22 @@ const InviteAddedToTeamNotice = ({
 type GitPushInfoProps = Props & {info: GitPushInfo}
 
 const GitPushInfoNotice = ({message, info}: GitPushInfoProps) => {
+  // There is a bug in the data layer where mergeEntities when it sees dupes of this message will keep on adding to the array
+  // Short term fix: clean this up
+
+  const refsMap = (info.refs || []).reduce((map, ref) => {
+    ;(ref.commits || []).forEach(commit => {
```

Comment:
Why not just assign `ref.commits` to `map[name]`? I.e. why is iteration necessary?

Reply:
the goal is to dedupe this, the data is a 2 level structure of 
```
 ref => [commits]
```

we don't know if we'll get dupe ref names w/ different commits. just trying to make no assumptions about the data right now

if we assign ref.commits to map[name] we'll overwrite data if we got
```
 ['ref1'] => [1,2], 
 ['ref1'] => [3,4]

```

right?

Reply:
Sure, but by appending to the existing commit list, we're communicating that those commits are part of the Merkle tree. I'd prefer to overwrite such that the tree is preserved.